### PR TITLE
Labster enrollment view

### PR DESCRIPTION
**Pull request by polesye**
This view helps automate enrollment workflow in the the new "direct to student" pricing experiments.
It's a copy of `ccx_invite` view with authentication adjustments (Token authentication is added).

@TamoshaytisV @lukyfebrianto please review. Could someone of you make sure tests pass on Dogwood release, because I didn't have a chance to run them locally?

@Bodekaer FYI.

### CODE REVIEW

**Comment by Bodekaer** [comment]
Awesome! Thanks so much @polesye!

In terms of activating this new `ccx_invite` feature, I've been thinking about the best way to do it.
I can do it via PHP on the WP marketing website, but I'm also thinking that it would be a better architecture if our API handled this.
We have the Stripe webhook setup now here:
https://github.com/Livit/Livit.Learn.LTI/blob/edd8b58d92a5ee338ed5985058f555abe32076e4/django_project/payments/views.py

Would it be possible to have it detect a new Stripe subscription, and then call the new `ccx_invite` EdX method? Or do you prefer to keep it in the Marketing site? 
You can see the Stripe `customer.subscription.created` event details here:
https://stripe.com/docs/api#event_types-customer.subscription.created

**Comment by Bodekaer** [comment]
We can just hardcode the mapping in the beginning.
E.g. if stripe subscription plan id = X, invite to CCX course Y.

Or alternatively make a simple `SubscriptionCourses` Django model/table with two fields: 
`stripeSubscriptionPlanId`, `course_key`.

**Comment by polesye** [comment]
@Bodekaer I'd prefer to keep it in the Marketing site atm, because it's more simple solution.

File: openedx/core/djangoapps/labster/course/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    def test_ccx_invite_enroll_up_to_limit(self):
+        """
+        Enrolls a list of students up to the enrollment limit.
+
+        This test is specific to one of the enrollment views: the reason is because
+        the view used in this test can perform bulk enrollments.
+        """
+        self.make_coach()
+        # create ccx and limit the maximum amount of students that can be enrolled to 2
+        ccx = self.make_ccx(max_students_allowed=2)
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/tests/test_views.py]: did we have max_students_allowed args on make_ccx ?

File: openedx/core/djangoapps/labster/course/views.py [hunk 2]
```diff
@@ -4,21 +4,27 @@
 from django.http import Http404
 from django.contrib.auth.models import User
 from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.cache import cache_control
 from rest_framework.decorators import api_view, authentication_classes, permission_classes
 from rest_framework.authentication import SessionAuthentication, TokenAuthentication
 from rest_framework.permissions import IsAuthenticated
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py]: remove this if unused

File: openedx/core/djangoapps/labster/course/views.py [hunk 3]
```diff
@@ -4,21 +4,27 @@
...
 from rest_framework.response import Response
 from rest_framework import status
 from rest_framework.reverse import reverse
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py]: remove this if unused

File: openedx/core/djangoapps/labster/course/views.py [hunk 4]
```diff
@@ -49,7 +55,7 @@ def setup_course(course_key, staff=None):
 @csrf_exempt
 @api_view(['GET', 'POST', 'PUT', 'DELETE'])
 @authentication_classes((SessionAuthentication, TokenAuthentication))
-@permission_classes((IsAuthenticated,))
+@permission_classes((IsAdminUser,))
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py:52]: do we have import func for IsAdminUser ?

File: openedx/core/djangoapps/labster/course/views.py [hunk 5]
```diff
@@ -147,7 +153,7 @@ def course_handler(request, course_key_string=None):
 
 @api_view(['GET'])
 @authentication_classes((SessionAuthentication, TokenAuthentication))
-@permission_classes((IsAuthenticated,))
+@permission_classes((IsAdminUser,))
 def coach_list_handler(reuqest):
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py]: reuqest or request ?

File: openedx/core/djangoapps/labster/course/views.py [hunk 6]
```diff
@@ -158,3 +164,68 @@ def coach_list_handler(reuqest):
...
+            If email_students is true, students will be sent email notification
+            If email_students is false, students will not be sent email notification
+    """
+    if not ccx:
+        raise Http404
+
+    action = request.POST.get('action')
+    identifiers_raw = request.POST.get('identifiers')
+    identifiers = _split_input_list(identifiers_raw)
+    email_students = 'email_students' in request.POST
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py]: do we need  this email_students ?

File: openedx/core/djangoapps/labster/course/views.py [hunk 7]
```diff
@@ -158,3 +164,68 @@ def coach_list_handler(reuqest):
...
     } for coach in coaches)
     return Response(coaches_info)
+
+
+@csrf_exempt
+@api_view(['POST'])
+@authentication_classes((SessionAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@coach_dashboard
```
Comment (lukyfebrianto) [inline openedx/core/djangoapps/labster/course/views.py]: do we have import for this decorator?

**Review by lukyfebrianto — COMMENTED** [review]
Please see my comment @polesye .CMIIW

File: openedx/core/djangoapps/labster/course/views.py (hunk 6 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py]: > do we need this email_students ?

Good catch. Removed.

File: openedx/core/djangoapps/labster/course/views.py (hunk 7 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py]: > do we have import for this decorator?

Typo, fixed.

File: openedx/core/djangoapps/labster/course/views.py (hunk 5 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py]: > reuqest or request ?

typo is fixed

File: openedx/core/djangoapps/labster/course/views.py (hunk 4 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py:52]: > do we have import func for IsAdminUser ?

Fixed

File: openedx/core/djangoapps/labster/course/views.py (hunk 3 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py]: > remove this if unused

Done

File: openedx/core/djangoapps/labster/course/views.py (hunk 2 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/views.py]: > remove this if unused

Done

File: openedx/core/djangoapps/labster/course/tests/test_views.py (hunk 1 above)
Comment (polesye) [reply openedx/core/djangoapps/labster/course/tests/test_views.py]: > did we have max_students_allowed args on make_ccx ?

Good point, extended.

**Comment by polesye** [comment]
@lukyfebrianto @TamoshaytisV I've addressed comments. Please continue review.

**Review by TamoshaytisV — CHANGES_REQUESTED** [review]
You've replaced
```
-        return CcxFactory(course_id=self.course.id, coach=self.user)
```
to
```
 +        ccx = CcxFactory(course_id=self.course.id, coach=self.coach)
```
but there is no self.coach class field

File: lms/djangoapps/labster_enroll/views.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+def ccx_invite(request, course_id):
+    """
+    Enroll or unenroll students to CCX by email.
+    Requires staff access.
+
+    **Request example**:
+        POST /labster/api/course/enroll
+        {
+            "course_key": "course-v1:labster+LABX+2015_T1",
```
Comment (polesye) [inline lms/djangoapps/labster_enroll/views.py]: remove this line, it's a part of url

File: lms/djangoapps/labster_enroll/views.py [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+@authentication_classes((SessionAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+def ccx_invite(request, course_id):
+    """
+    Enroll or unenroll students to CCX by email.
+    Requires staff access.
+
+    **Request example**:
+        POST /labster/api/course/enroll
```
Comment (polesye) [inline lms/djangoapps/labster_enroll/views.py]: please fix url format, it should include course key

**Comment by TamoshaytisV** [comment]
@lukyfebrianto @polesye please finish review

**Review by TamoshaytisV — APPROVED** [review]

**Review by lukyfebrianto — APPROVED** [review]
Looks good to me

**Pull request merged**